texastribune/donations
# Integrate design system build tasks

#### What's this PR do?

Compiles SCSS into a `static/css` and builds the base SVG sprite for later icon usage.

#### Why are we doing this? How does it help us?

Shifts our CSS/iconography config to one shared build system. For example, when we want to make changes to autoprefixer or any other post CSS setting we do it in [the toolbox](https://github.com/texastribune/ds-toolbox) and the styles in this repo will inherit those upgrades. In the future our other repos will also inherit them.

**TL;DR one source of truth for the front-end (sans JS) build** 

#### How should this be manually tested?

Run a fresh `yarn`

Run `yarn run dev`
 - Check for `Success! Watching styles...` and make sure webpack is still doin' its thing.
 - Break the SCSS by commenting out an important variable like `$color-yellow-tribune` or something. Make sure it barks at you. Webpack will too.

Question that came up in this: _Would it be correct to say `yarn run dev` does *not* start the app server anymore? And for this we now rely on `make restart`?_

Run `yarn run postinstall`
 - Check for `Success!` and make sure webpack is still doin' its thing.
 - Break the SCSS by commenting out an important variable like `$color-yellow-tribune` or something. Make sure it barks at you. It won't get to webpack bc it exits with `process.exit(1);`.

 
#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

We'll want to switch the devDependency from `https://github.com/texastribune/ds-toolbox.git#dependencies` to `https://github.com/texastribune/ds-toolbox.git#master` once this is approved.

#### What are the relevant tickets?

No specific task, but in the UMP world, this tees us up for UMP using the design system 🎉 

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***

* [ ] Updated the documentation/wiki? *(will do once we hammer out the details )*

#### TODOs / next steps:

* [ ] *Hook up the styles to the new location*
* [ ] *Start using the style base from ds-toolbox (lonnng term, but UMP could!)*
* [ ] *Remove webpacks CSS importing? Maybe this deserves a discussion.*
* [ ] *Start using SVG sprite by overhauling font-awesome like we did in our main repo*
